### PR TITLE
[stable/jenkins] add nodes

### PR DESCRIPTION
**Pull request by liangsdet**
Signed-off-by: liangsdet <sdetliang@gmail.com>

#### What this PR does / why we need it:
Add Jenkins node config

#### Special notes for your reviewer:
Thank you

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by liangsdet** [comment]
@lachie83 
Could you please take a look?

**Comment by liangsdet** [comment]
Hi @viglesiasce, @lachie83 and @mattfarina
Could you please take a look at this?

File: stable/jenkins/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: jenkins
 home: https://jenkins.io/
-version: 0.25.0
+
```
Comment (viglesiasce) [inline stable/jenkins/Chart.yaml]: Remove whitespace.

File: stable/jenkins/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+version: 0.26.0
+
```
Comment (viglesiasce) [inline stable/jenkins/Chart.yaml]: remove whitespace

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Can you add documentation to the README on how/why this is used?

File: stable/jenkins/Chart.yaml (hunk 1 above)
Comment (liangsdet) [reply stable/jenkins/Chart.yaml]: Thanks @viglesiasce , It's fixed

File: stable/jenkins/Chart.yaml (hunk 2 above)
Comment (liangsdet) [reply stable/jenkins/Chart.yaml]: Thanks @viglesiasce , It's fixed

**Comment by liangsdet** [comment]
Hi @viglesiasce,

Thanks for the review, the issue in review has been fixed. And the instruction has been added into README.

Could you please review again? This new feature will add permanent nodes.

**Comment by viglesiasce** [comment]
Thanks @liangsdet! 

Adding @maorfr who is helping out with Jenkins chart reviews.

@maorfr can you take a look?

/unassign
/assign @maorfr

**Comment by liangsdet** [comment]
@maorfr 
Thanks a lot for your reply. I have resolve the conflict and add the following use case.

`If you need to add dedicated and permanent agent machine, you may need this.`

**Comment by maorfr** [comment]
Thanks @liangsdet,

First of all, I can't install the chart with your changes -
```
MountVolume.SetUp failed for volume "jenkins-nodes" : configmap "jenkins-nodes" not found
```

I would like to better understand this feature.
When you say nodes - you actually mean slaves. Am I correct?

If so, since we already have a slave configuration under `Agent` in the values file, and a `config.yaml` template to describe it, it feels strange to define the xml for a node again (duplication?).

If I am indeed correct, I would expect this feature to be exposed in a very simple way. For example:
values.yaml:
```
...
Agent:
  PermanentNodes: 0 # change this setting to have permanent nodes
```

In addition, I don't feel it is right to define the node spec in `values.yaml`, it is cumbersome enough as it is.

Would love to hear your thoughts on that. If I am mistaken, please correct me!

**Comment by liangsdet** [comment]
Thanks, will close for now

**Pull request closed without merging**